spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractConsumerSeekAware.java

```diff
@@ -46,43 +47,59 @@ public abstract class AbstractConsumerSeekAware implements ConsumerSeekAware {
 
 	@Override
 	public void registerSeekCallback(ConsumerSeekCallback callback) {
-		this.callbackForThread.put(Thread.currentThread(), callback);
+		if (matchGroupId()) {
+			this.callbackForThread.put(Thread.currentThread(), callback);
```

Comment:
I'm sorry, I don't see how is this logic different from what you have suggested as an option on the `@KafkaListener`?
I still don't see `groupId` matching though...

Reply:
You're right. Using a flag to control the behavior is similar. The main difference is that the PR approach don't use the top-level property of @KafkaListener. It avoids the somewhat awkward and unrelated task of having @KafkaListener determine whether to use seek. Instead, the decision of whether to apply seek by group id has been moved to `ConsumerSeekAware`, which is more fitting for this role.

Reply:
And `groupId` can be matched like [this](https://github.com/bky373/spring-kafka/commit/b93ff92a4e2e2ec69329783c7bc515494062cd21#diff-d59d8f0f475e848f3ab4a6d6768c513aa1e6b45d61a762cec1ff50fc39164980R107). Am I misunderstanding something?

Reply:
Oh! That is smart!
I think such a logic for the `KafkaUtils.getConsumerGroupId()` has to be hidden from end-user and its result propagated to that `matchGroupId()` method impl.
Or do you have any other use-case in mind?

Thank you for explanation!

Reply:
That's a great point, Artem. Ya, the end users should not worry about calling `KafkaUtils.getConsumerGroupId()`.

Reply:
Thanks! It seems like a good idea to hide `KafkaUtils.getConsumerGroupId()` from the user. It's late here, so I'll try to work on this tomorrow or by the end of today.